File: src/namedparams/NamedParameters.java
package namedparams;

/**
 * @author Sergey Ryabov <colriot@gmail.com>
 *         20/06/2017
 */
public class NamedParameters {

  private static void composePerson(String firstName, String lastName, int age, int shoeSize,
      double height, double weight, boolean isVegetarian) {
    // Some usefull code here.
  }

  public static void main(String[] args) {
    composePerson("Hamato", "Yoshi", 42, 39, 149.5, 72.0, true);
  }
}

File: src/namedparams/NamedParameters.kt
package namedparams

/**
 * @author Sergey Ryabov <colriot@gmail.com>
 * 20/06/2017
 */
private fun composePerson(firstName: String, lastName: String, age: Int, shoeSize: Int,
                          height: Double, weight: Double, isVegetarian: Boolean) {
  // Some usefull code here.
}

fun main(args: Array<String>) {
  composePerson(
      firstName = "Hamato",
      lastName = "Yoshi",
      age = 42,
      shoeSize = 39,
      height = 149.5,
      weight = 72.0,
      isVegetarian = true
  )
}
